x/qgb/client/tx.go

```diff
@@ -0,0 +1,56 @@
+package client
+
+import (
+	"fmt"
+
+	"github.com/celestiaorg/celestia-app/x/qgb/types"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	"github.com/spf13/cobra"
+)
+
+// GetTxCmd returns the transaction commands for this module
+func GetTxCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Short:                      fmt.Sprintf("%s transactions subcommands", types.ModuleName),
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+
+	cmd.AddCommand(CmdRegisterEVMAddress())
+
+	return cmd
+}
+
+func CmdRegisterEVMAddress() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "register [valAddress] [evmAddress]",
+		Short: "Register an EVM address for a validator",
+		Long: `Registers an EVM address for a validator. This address will be used to
+sign attestations as part of the QGB protocol. Only the validator, as the signer
+can register an EVM address. To change the EVM address, the validator can simply
+send a new message overriding the previous one.
+`,
+		Args: cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			valAddress, evmAddress := args[0], args[1]
```

Comment:
Shouldn't we do more validation on the arguments at this level? so that a user doesn't end up sending a transaction and be a failed one because some address is not well formatted

Reply:
`NewMsgRegisterEVMAddress` below should sanitize the input (i.e. check that it is a ValAddress and a HexAddress)

Reply:
Let me know if you think there is more validation that should be done i.e. maybe check size of the evm address

Reply:
that's the only thing I guess